src/parser.yy

```diff
@@ -215,7 +215,9 @@ ternary : expr QUES expr COLON expr { $$ = new ast::Ternary($1, $3, $5, @$); }
 
 param : PARAM      {
                      try {
-                       $$ = new ast::PositionalParameter(PositionalParameterType::positional, std::stol($1.substr(1, $1.size()-1)), @$);
+                       long n = std::stol($1.substr(1, $1.size()-1));
+                       if (n == 0) throw std::exception();
```

Comment:
Why throw an exception over calling `error(...); YYERROR`?

Reply:
The intention is to let the catch block do the work.

https://github.com/iovisor/bpftrace/blob/a03b5c4bbb12a25345a2fdec78374963ff936131/src/parser.yy#L217-L223

But I don't have a strong opinion, which do you prefer?

Reply:
Ah I didn't notice that line was already in a try/catch block. It's fine then